src/common/utxobased/network/Socket.ts

```diff
@@ -271,7 +277,11 @@ export function makeSocket(uri: string, config: SocketConfig): Socket {
             if (!subscription.subscribed) {
               subscription.deferred.reject()
             }
-            subscription.cb(json.data)
+            if (subscription.cleaner != null) {
+              subscription.cb(subscription.cleaner(json.data))
+            } else {
+              subscription.cb(json.data)
+            }
```

Comment:
No need for null case check because cleaner is not optional on the subscription object type.